zk_toolbox/crates/zk_inception/src/commands/chain/migrate_to_gateway.rs

```diff
@@ -0,0 +1,465 @@
+        l1_url.clone(),
+    )
+    .await?;
+    println!(
+        "operator supplied with 10 ETH! Hash: {}",
+        hex::encode(hash.as_bytes())
+    );
+
+    let gateway_url = gateway_chain_config
+        .get_general_config()
+        .unwrap()
+        .api_config
+        .unwrap()
+        .web3_json_rpc
+        .http_url
+        .clone();
+
+    let mut chain_secrets_config = chain_config.get_secrets_config().unwrap();
+    chain_secrets_config.l1.as_mut().unwrap().gateway_url =
+        Some(url::Url::parse(&gateway_url).unwrap().into());
+    chain_secrets_config.save_with_base_path(shell, chain_config.configs.clone())?;
+
+    let gateway_chain_config = GatewayChainConfig::from_gateway_and_chain_data(
+        &gateway_gateway_config,
+        new_diamond_proxy_address,
+        // TODO: for now we do not use a noraml chain admin
+        Address::zero(),
+    );
+    gateway_chain_config.save_with_base_path(shell, chain_config.configs.clone())?;
+
+    let mut general_config = chain_config.get_general_config().unwrap();
+
+    let eth_config = general_config.eth.as_mut().expect("eth");
+    let api_config = general_config.api_config.as_mut().expect("api config");
+
+    eth_config
+        .gas_adjuster
+        .as_mut()
+        .expect("gas_adjuster")
+        .settlement_mode = SettlementMode::Gateway;
+    // FIXME: for now only rollups are supported
+    eth_config
+        .sender
+        .as_mut()
+        .expect("sender")
+        .pubdata_sending_mode = PubdataSendingMode::RelayedL2Calldata;
+    eth_config
+        .sender
+        .as_mut()
+        .context("sender")?
+        .wait_confirmations = Some(0);
+    // FIXME: do we need to move the following to be u64?
+    eth_config
+        .sender
+        .as_mut()
+        .expect("sender")
+        .max_aggregated_tx_gas = 4294967295;
+    api_config.web3_json_rpc.settlement_layer_url = Some(gateway_url);
+    general_config.save_with_base_path(shell, chain_config.configs.clone())?;
+
+    let mut chain_genesis_config = chain_config.get_genesis_config().unwrap();
+    chain_genesis_config.sl_chain_id = Some(gateway_chain_id.into());
+    chain_genesis_config.save_with_base_path(shell, chain_config.configs.clone())?;
+
+    Ok(())
+}
+
+async fn await_for_tx_to_complete(
+    gateway_provider: &Provider<Http>,
+    hash: H256,
+) -> anyhow::Result<()> {
+    println!("Waiting for transaction to complete...");
+    while gateway_provider
+        .get_transaction_receipt(hash)
+        .await?
+        .is_none()
+    {
+        tokio::time::sleep(tokio::time::Duration::from_millis(200)).await;
+    }
+
+    // We do not handle network errors
+    let receipt = gateway_provider
+        .get_transaction_receipt(hash)
+        .await?
+        .unwrap();
+
+    if receipt.status == Some(U64::from(1)) {
+        println!("Transaction completed successfully!");
+    } else {
+        panic!("Transaction failed!");
+    }
+
+    Ok(())
+}
+
+async fn call_script(
```

Comment:
This function is duplicated, probably ok for now

Reply:
let's keep it this way for now, I have a feeling that Danil would want to refactor lots of stuff here anywhere